exchanges/newexchange.md

```diff
@@ -0,0 +1,766 @@
+}
+```
+
+Get Request params for authenticated requests are sent through url.Values{}:
+
+https://docs.ftx.com/#get-withdrawal-history:
+
+```go
+// GetTriggerOrderHistory gets trigger orders that are currently open
+func (f *FTX) GetTriggerOrderHistory(marketName string, startTime, endTime time.Time, side, orderType, limit string) (TriggerOrderHistory, error) {
+	var resp TriggerOrderHistory
+	params := url.Values{}
+	if marketName != "" {
+		params.Set("market", marketName)
+	}
+	if !startTime.IsZero() && !endTime.IsZero() {
+		params.Set("start_time", strconv.FormatInt(startTime.Unix(), 10))
+		params.Set("end_time", strconv.FormatInt(endTime.Unix(), 10))
+		if !startTime.Before(endTime) {
+			return resp, errors.New("startTime cannot be bigger than endTime")
+		}
+	}
+	if side != "" {
+		params.Set("side", side)
+	}
+	if orderType != "" {
+		params.Set("type", orderType)
+	}
+	if limit != "" {
+		params.Set("limit", limit)
+	}
+	return resp, f.SendAuthHTTPRequest(http.MethodGet, getTriggerOrderHistory+params.Encode(), nil, &resp)
+}
+```
+
+For post or delete requests params are sent through map[string]interface{}:
+
+https://docs.ftx.com/#place-order
+
+
+Structs for unmarshalling the data are made exactly the same way as the previous functions
+```go
+type OrderData struct {
+	CreatedAt     time.Time `json:"createdAt"`
+	FilledSize    float64   `json:"filledSize"`
+	Future        string    `json:"future"`
+	ID            int64     `json:"id"`
+	Market        string    `json:"market"`
+	Price         float64   `json:"price"`
+	AvgFillPrice  float64   `json:"avgFillPrice"`
+	RemainingSize float64   `json:"remainingSize"`
+	Side          string    `json:"side"`
+	Size          float64   `json:"size"`
+	Status        string    `json:"status"`
+	OrderType     string    `json:"type"`
+	ReduceOnly    bool      `json:"reduceOnly"`
+	IOC           bool      `json:"ioc"`
+	PostOnly      bool      `json:"postOnly"`
+	ClientID      string    `json:"clientId"`
+}
+
+// PlaceOrder stores data of placed orders
+type PlaceOrder struct {
+	Success bool      `json:"success"`
+	Result  OrderData `json:"result"`
+}
+```
+
+```go
+// Order places an order
+func (f *FTX) Order(marketName, side, orderType, reduceOnly, ioc, postOnly, clientID string, price, size float64) (PlaceOrder, error) {
+	req := make(map[string]interface{})
+	req["market"] = marketName
+	req["side"] = side
+	req["price"] = price
+	req["type"] = orderType
+	req["size"] = size
+	if reduceOnly != "" {
+		req["reduceOnly"] = reduceOnly
+	}
+	if ioc != "" {
+		req["ioc"] = ioc
+	}
+	if postOnly != "" {
+		req["postOnly"] = postOnly
+	}
+	if clientID != "" {
+		req["clientID"] = clientID
+	}
+	var resp PlaceOrder
+	return resp, f.SendAuthHTTPRequest(http.MethodPost, placeOrder, req, &resp)
+}
+```
+
++ 6) Implementing wrapper functions:
+
+Wrapper functions are the interface through which GCT bot communicates with exchange for gathering and sending data
```

Comment:
Wrapper functions are the interface through which GCT bot communicates with an exchange for gathering and sending data.